### DISCUSSION TITLE
Mensaje de error generado al presentar la tarea al calificador - Task 2

### DISCUSSION THREAD

**Discussion by maguiargil** [rramosp/ai4eng.v1#227] [03 - GENERACION DE MODELOS]
Cordial saludo,

En la tarea 2 (task 2) estoy recibiendo este mensaje de error al presentarle al calificador mi propuesta de solución:

![image](https://user-images.githubusercontent.com/55927354/220448103-b31276de-87db-4cae-b273-c2f1e1ee7312.png)

Me pregunto por la naturaleza de dicho mensaje de error. Muchas gracias por la atención, de antemano agradezco la respuesta que me brinden.

**Comment by Felipe-RA** [comment]
Saludos @maguiargil 

Tu error es un `NameError`, que ocurre cuando usas un nombre no definido en tu código.

En tu caso el nombre es `pd`, que probablemente lo tienes definido **fuera** de la función que califica el grader.

Para resolver este error, importa todas las librerias que necesites dentro de tu función.

¡Espero te ayude, Quedo pendiente!

**Reply by maguiargil** [reply]
Cordial saludo,

"pd" es el alias de la librería Pandas, este alias ya viene especificado en  las primeras líneas de todo el notebook de trabajo que estoy usando, incluso esta importación es literalmente una copia de las importaciones que aparecen en la guía del laboratorio. Estando ya declarado y ejecutado este alias, no veo por qué aún arroje  dicho error.

**Reply by Felipe-RA** [reply]
Efectivamente maguiargil, pd es el alias de pandas.

El entorno de los graders es distinto a tu notebook, es por eso que debes de incluir las librerías dentro de tu función, para que el grader pueda tomarla y saber qué librerias debe importar.

Puedes pensar la diferencia de `scope` (el alcance de la función) como algo similar a este gráfico:

<img width="942" alt="image" src="https://user-images.githubusercontent.com/49454068/220460765-e88544b0-9dc6-4b3b-b7ca-1aba1660b847.png">

**Reply by maguiargil** [reply]
Ya lo tengo en mi notebook, leyendo la salida del mensaje de error intuyo (tal vez esté equivocado) que el error está en alguno de los archivos dentro del servidor de calificación.

[En este enlace se encuentra mi notebook para la revisión](https://colab.research.google.com/drive/1f02cSIQTokMuE1NfFgaA0mru-R8j1h-3?usp=sharing)

**Reply by Felipe-RA** [reply]
A lo que me refiero es que debes de incluir el import de pandas en tu función fit. Lo que se manda es tu función, no todo el contenido de tu notebook.

Veo en tu notebook que no lo has incluido.

Tu función debe ser autosuficiente, el error ocurre en el servidor de calificación porque intenta correr tu función pero no encuentra el nombre `pd`. Debes incluir y declarar todo lo que vayas a utilizar dentro de la función que se está evaluando

**Reply by maguiargil** [reply]
Bueno, ahora sí corrió. No sabía que el calificador tomaba solo algunas celdas de código. Muchas gracias por la corrección.

**Reply by Felipe-RA** [reply]
Me alegra que ya te funcione!

Pero recuerda, no es que tome sólo algunas celdas.

Toma sólo algunas funciones y/o variables, usualmente listadas en la descripción del ejercicio.
